src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -47,43 +53,61 @@ public override void Initialize(IHostContext hostContext)
             _agentServer = HostContext.GetService<IAgentServer>();
         }
 
-        public void UpdateAgentSetting(AgentSettings settings)
+        public AgentConfigSettings ReadSettingsAndGetAgentConfigSettings(AgentSettings settings)
+        {
+            return new AgentConfigSettings(settings.PoolId);
+        }
+
+        public void UpdateAgentSetting(AgentConfigSettings agentConfigSettings, AgentSettings settings)
         {
-            // No implementation required
         }
 
         public string GetServerUrl(CommandSettings command)
         {
             return command.GetUrl();
         }
 
-        public async Task<int> GetPoolId(CommandSettings command)
+        public async Task<AgentConfigSettings> GetAgentConfigSettings(CommandSettings command)
         {
             int poolId = 0;
             string poolName;
 
             poolName = command.GetPool();
             poolId = await GetPoolIdAsync(poolName);
             Trace.Info($"PoolId for agent pool '{poolName}' is '{poolId}'.");
- 
-            return poolId;
+
+            return new AgentConfigSettings(poolId);
         }
 
         public string GetFailedToFindPoolErrorString() => StringUtil.Loc("FailedToFindPool");
 
-        public Task<TaskAgent> UpdateAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public string GetAgentWithSameNameAlreadyExistErrorString(AgentConfigSettings agentConfigSettings, string agentName)
         {
-            return _agentServer.UpdateAgentAsync(poolId, agent);
+            return StringUtil.Loc("AgentWithSameNameAlreadyExistInPool", agentConfigSettings.PoolId, agentName);
         }
 
-        public Task<TaskAgent> AddAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public Task<TaskAgent> UpdateAgentAsync(AgentConfigSettings agentConfigSettings, TaskAgent agent, CommandSettings command)
+        {
+            return _agentServer.UpdateAgentAsync(agentConfigSettings.PoolId, agent);
+        }
+
+        public Task<TaskAgent> AddAgentAsync(AgentConfigSettings agentConfigSettings, TaskAgent agent, CommandSettings command)
         { 
-            return _agentServer.AddAgentAsync(poolId, agent);
+            return _agentServer.AddAgentAsync(agentConfigSettings.PoolId, agent);
         }
 
-        public Task DeleteAgentAsync(int agentPoolId, int agentId)
+        public async Task DeleteAgentAsync(AgentConfigSettings agentConfigSettings, AgentSettings settings, string currentAction)
         {
-            return _agentServer.DeleteAgentAsync(agentPoolId, agentId);
+            TaskAgent agent = await GetAgentAsync(agentConfigSettings, settings.AgentName);
+            if (agent == null)
+            {
+                _term.WriteLine(StringUtil.Loc("Skipping") + currentAction);
```

Comment:
this gives error like agent not found skipping.  nothing changed from existing behavior apart from moved code form manger to provider :)